File: algo/code-js/02-02求和.js
// 求和 计算不大于n的项的总和
// 1 + 4 + 10 + ... 
// 后一项 = (前一项 + 1) * 2


function getSum(n) {
    let sum = 0
    let i = 1
    while (i <= n) {
        sum += i
        i ++
        i *= 2
    }
    return sum
}

// 调用函数
let n = 5
let result = getSum(n)
console.log(result) // 5

File: algo/code/02/py/02-02求和.py
"""求和 计算不大于n的项的总和
1 + 4 + 10 + ... 
后一项 = (前一项 + 1) * 2
"""

def get_sum(n: int) -> int:
    sum = 0
    i = 1
    while i <= n:
        sum += i
        i += 1
        i *= 2
    return sum

if __name__ == '__main__':
    n = 5
    print(get_sum(n)) # 5